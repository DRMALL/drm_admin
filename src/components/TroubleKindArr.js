

import React from 'react'
import troubleKindToTop from '../actions/troubleKindToTop'
import troubleKindDel from '../actions/troubleKindDel'
import Popconfirm from 'antd/lib/popconfirm'
import dispatch from '../actions/dispatch'


export default  ({ troubleKinds })=> (
    <div>
      {
        troubleKinds.map( (item, index) =>
        <div key={ item._id } className='trouble-kind-arr'  >
          <div className='troubel-kind-left' >{ item.text }</div>
          <i className='iconfont icon-zhiding troubel-kind-mid'
             data-id={ item._id }  style={{ visibility: index===0 ? 'hidden' : 'visible' }}
             onClick={ troubleKindToTop } ></i>
          <Popconfirm title='确定删除本条分类? 删除后本类下的内容将归于其他'
                      onConfirm={ troubleKindDel }
                      okText="确定" cancelText="取消"
                      data-id={ item._id } >
            <i className='iconfont icon-shanchu troubel-kind-right'
               data-id={ item._id }
               onClick={ getKindId }  ></i>
          </Popconfirm>
        </div> )
      }
    </div>
  )

function getKindId(e){
  dispatch('TROUBLE_GET_TROUBLE_KIND_ID', e.currentTarget.dataset.id)
}

// function confirmDelKind(e){
//     console.log(e.currentTarget.dataset)
//     // onClick={ troubleKindDel }
// }

