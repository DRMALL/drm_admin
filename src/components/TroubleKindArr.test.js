

import React from 'react'
import TroubleKindArr from './TroubleKindArr'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

it('TroubleKindArr test', () => {
  const troubleKinds = [{_id: 'sd4225', text: 'swdafrt text'},{_id: 'sd4sd25', text: 'swdafrt text'}]
  const wrapper = render(<TroubleKindArr troubleKinds= {troubleKinds} /> )
  expect(toJson(wrapper)).toMatchSnapshot()
})
