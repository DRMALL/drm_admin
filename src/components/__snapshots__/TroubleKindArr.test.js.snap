// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TroubleKindArr test 1`] = `
<div>
  <div
    class="trouble-kind-arr"
  >
    <div
      class="troubel-kind-left"
    >
      swdafrt text
    </div>
    <i
      class="iconfont icon-zhiding troubel-kind-mid"
      data-id="sd4225"
      style="visibility:hidden;"
    />
    <i
      class="iconfont icon-shanchu troubel-kind-right"
      data-id="sd4225"
    />
  </div>
  <div
    class="trouble-kind-arr"
  >
    <div
      class="troubel-kind-left"
    >
      swdafrt text
    </div>
    <i
      class="iconfont icon-zhiding troubel-kind-mid"
      data-id="sd4sd25"
      style="visibility:visible;"
    />
    <i
      class="iconfont icon-shanchu troubel-kind-right"
      data-id="sd4sd25"
    />
  </div>
</div>
`;
